pymc_marketing/mlflow.py

```diff
@@ -412,14 +460,319 @@ def log_inference_data(
+                "event_2",
+                "t",
+            ],
+            yearly_seasonality=2,
+        )
+
+        mlflow.set_experiment("MMM Experiment")
+
+        with mlflow.start_run():
+            idata = mmm.fit(X, y)
+
+            # Additional specific logging
+            fig = mmm.plot_components_contributions()
+            mlflow.log_figure(fig, "components.png")
+
+            model_info = log_model(mmm,
+                    mmm_registry_name="my_amazing_mmm",
+                    include_last_observations=True,
+                    original_scale=False,
+                    )
+    """
+    # Incorporate MMM into MLflow workflow
+    mlflow_mmm = MMMWrapper(
+        mmm,
+        extend_idata=wrapper_kwargs.get("extend_idata", False),
+        combined=wrapper_kwargs.get("combined", True),
+        include_last_observations=wrapper_kwargs.get(
+            "include_last_observations", False
+        ),
+        original_scale=wrapper_kwargs.get("original_scale", True),
+    )
+
+    mlflow.pyfunc.log_model(
+        artifact_path=artifact_path,
+        python_model=mlflow_mmm,
+    )
+    run_id = mlflow.active_run().info.run_id
+    model_uri = f"runs:/{run_id}/{artifact_path}"
+
+    if mmm_registry_name:
+        mlflow.register_model(model_uri, mmm_registry_name)
+
+
+def load_model(
+    run_id: str | None = None,
+    full_model: bool = False,
+    keep_idata: bool = False,
+    artifact_path: str = "model",
+) -> mlflow.pyfunc.PyFuncModel | MMM:
+    """
+    Load a PyMC-Marketing MMM model from MLflow.
+
+    Can either load the full model including the InferenceData, or just the lighter PyFuncModel version.
+
+    Parameters
+    ----------
+    run_id : str, optional
+        The MLflow run ID from which to load the model.
+    full_model : bool, default=True
+        If True, load the full MMM model including the InferenceData.
+    keep_idata : bool, default=False
+        If True, keep the downloaded InferenceData saved locally.
+    artifact_path : str, default="model"
+        The artifact path within the run where the model is stored.
+
+    Returns
+    -------
+    model : mlflow.pyfunc.PyFuncModel | MMM
+        The loaded MLflow PyFuncModel or MMM model.
+
+
+    Examples
+    --------
+    # Load model using run_id
+    model = load_model(run_id="your_run_id", full_model=True, keep_idata=True)
+    """
+    model_uri = f"runs:/{run_id}/{artifact_path}"
+
+    if not full_model:
+        model = mlflow.pyfunc.load_model(model_uri)
+
+    else:
+        idata_path = mlflow.artifacts.download_artifacts(
+            run_id=run_id, artifact_path="idata.nc", dst_path="idata"
+        )
+        model = MMM.load(idata_path)
+        if not keep_idata:
+            os.remove(idata_path)
+            os.rmdir("idata")
+
+    return model
+
+
 @autologging_integration(FLAVOR_NAME)
 def autolog(
     log_sampler_info: bool = True,
-    log_datasets: bool = True,
+    log_metadata_info: bool = True,
```

Comment:
I find "_info" redundant with name of metadata

Reply:
But isn't it necessary otherwise you're passing an arg which is also the name of a function so it throws an error? Is there a name for the arg you like better?

Reply:
We'd have to separate out into module. However, we can wait on that